### DISCUSSION TITLE
Large middleware size

### DISCUSSION THREAD

**Discussion by arlyon** [nextauthjs/next-auth#5961] [Ideas]
### Environment

  System:
    OS: Linux 6.0 Fedora Linux 37 (Workstation Edition)
    CPU: (16) x64 AMD Ryzen 7 PRO 5850U with Radeon Graphics
    Memory: 18.20 GB / 27.23 GB
    Container: Yes
    Shell: 5.9 - /usr/bin/zsh
  Binaries:
    Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.14.2/bin/yarn
    npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
    Watchman: 0.0.0 - /usr/bin/watchman
  Browsers:
    Chrome: beta
    Firefox: 107.0
  npmPackages:
    next: latest => 13.0.6 
    next-auth: latest => 4.18.0 
    react: ^18.2.0 => 18.2.0 

### Reproduction URL

https://github.com/nextauthjs/next-auth-example

### Describe the issue

The default out-of-the-box middleware is about 50KB:

```
| conf  | size   |
| ----- | ------ |
| none  | 0KB    |
| empty | 13.8KB |
| auth  | 62.6KB |
```

This seems oddly _large_ to me given that the entirety of react + react-dom + nextjs + next-auth comes in at 83.5KB in the same example project.

### How to reproduce

Build the example project.

### Expected behavior

Smaller bundle size.

**Comment by Kinbaum** [comment]
@balazsorban44 Has the been any further investigation into this?
I am running into the same issue with a vanilla create-next-app using app dir.

My real application uses i18n and follows the official Next.js example on how to implement this with middleware.
I have some routes I am protecting as well. This is when I saw the large file size.

So I stripped back my entire middleware file to be just the following content:
```ts
// middleware.ts
export { default } from 'next-auth/middleware';

export const config = {
  matcher: [ '/admin' ]
};
```

This produces the following size when running `next build`

```diff
- ƒ Middleware                               213 kB
```

If I omit all of the next-auth middleware-related items, and just use middleware for the locale checks, I get the following size:

```diff
+ ƒ Middleware                               29.2 kB
```

**Comment by alexnguyennz** [comment]
Hi is there any update on this?

**Reply by ScareCrow95** [reply]
Just ran into the exact same issue :(

**Reply by Chris-Sahyouni** [reply]
same here
